### DISCUSSION TITLE
Auxiliary function to model forced events on a state variable.

### DISCUSSION THREAD

**Discussion by vassilismin** [lululxvi/deepxde#1476] [Q&A]
Hello!
First of all, congrats for the development of DeepXDE and the support you provide.

I'm currently working for my MSc thesis and I'm  trying to estimate the parameters of a pharmacokinetics model (inverse problem) that consists of a relatively large set of ODEs. However, I'm trying to use an auxiliary function in order to add specific amounts at a state variable at specific time points (it is like administering multiple doses of a drug to an organism). So, given that i have the data of the added amount and the time points, here is the auxiliary function i use: 

```
events_data = pd.read_csv('events_simplified.csv')
time_points = np.vstack(events_data['time'])
added_amount = np.vstack(events_data['value'])

def ex_func(t): 
    added_pfas = []
    for time_i in t:
        if np.isin(time_i, time_points):
            index = np.where(np.isin(time_points, time_i))
            added_pfas.append(added_amount[index])
        else:
            added_pfas.append(0)
    return(np.vstack(added_pfas))
```

 So, the unique input variable is time `t`, and if `t` is one of the values in `time_points`, then `added_pfas = added_amount else added_pfas = 0`. Note that `time_points` are later used as anchors.
 
I'm wondering if this style of modeling is causing any issues in the minimization of the ODEs gradients loss.
Thank you in advance. If you need me to provide any more code, I will be happy to help.

**Comment by lululxvi** [comment]
The arbitrary "0" may hurt your accuracy.
